include/xtensor/xtensor.hpp

```diff
@@ -548,6 +549,96 @@ namespace xt
 
         return out;
     };
+
+    /**
+     * Converts ``std::vector<index_type>`` (returned e.g. from ``xt::argwhere``) to a flattened
+     * ``xtensor``.
+     *
+     * @param vector of indices
+     *
+     * @return ``xt::xtensor<typename index_type::value_type, 1>`` (e.g. ``xt::xtensor<size_t, 1>``)
+     */
+    template <class T>
+    inline auto flatten_indices(const std::vector<T> &idx)
+    {
+        auto n = idx.size();
+        if (n != 0)
+        {
+            n *= idx[0].size();
+        }
+
+        using return_type = xtensor<typename T::value_type, 1>;
+        return_type out = xtensor<typename T::value_type, 1>::from_shape({n});
+        auto iter = out.begin();
+        for_each(idx.begin(), idx.end(), [&iter](const auto& t) { iter = std::copy(t.cbegin(), t.cend(), iter); });
+
+        return out;
+    };
+
+    struct ravel_vector_tag;
+    struct ravel_tensor_tag;
+
+    template <class C, class Tag>
+    struct ravel_return_type;
+
+    template <class C>
+    struct ravel_return_type<C, ravel_vector_tag>
+    {
+        using index_type = typename C::value_type;
+        using value_type = typename index_type::value_type;
+        using type = std::vector<value_type>;
+
+        template <class T>
+        static std::vector<value_type> init(T n)
+        {
+            return std::vector<value_type>(n);
+        }
+    };
+
+    template <class C>
+    struct ravel_return_type<C, ravel_tensor_tag>
+    {
+        using index_type = typename C::value_type;
+        using value_type = typename index_type::value_type;
+        using type = xt::xtensor<value_type, 1>;
+
+        template <class T>
+        static xt::xtensor<value_type, 1> init(T n)
+        {
+            return xtensor<value_type, 1>::from_shape({n});
+        }
+    };
+
+    template <class C, class Tag>
+    using ravel_return_type_t = typename ravel_return_type<C, Tag>::type;
```

Comment:
`reval_return_type_t` and the `ravel_return_type` specializations should be in a `detail` namespace so it's clear they're not part of the public API.